tests/cover/README.rst

```diff
@@ -0,0 +1,46 @@
+Yeah, this is the big one
+=========================
+
+This directory alone has around two-thirds of the tests for Hypothesis
+(~8k of ~12k lines of code).  If you're adding or fixing tests, chances
+are therefore good that they're in here!
+
+We require every change to come with tests in this directory that ensure
+100% branch coverage.  This can be an intimidating target, but it's entirely
+acheivable and this document is here to help you out.
+
+.. note::
+    If you get stuck, just ask a maintainer to help out by mentioning
+    ``@HypothesisWorks/hypothesis-python-contributors`` on GitHub``.
+    We'd love to help - and also get feedback on how this document could
+    be better!
+
+Some scenarios
+--------------
+
+**I'm adding or changing a strategy**
+    Check for a file specific to that strategy (eg ``test_uuids.py`` for
+    the ``uuids()`` strategy).  Write tests for all invalid argument handling
+    in ``test_direct_strategies.py``.  Strategies with optional dependencies
+    should go in ``hypothesis.extras``, and the tests in their own module
+    (ie not in ``cover``).  When you think you might be done, push and let
+    Travis point out any failing tests or non-covered code!
+
+**I've made some internal changes**
+    That's not very specific - you should probably refer to the test-finding
+    tips in the next section.  Remember that ``tests/cover`` is reasonably
+    quick unit-test style tests - you should consider writing more intensive
+    integration tests too, but put them in ``tests/nocover`` with the others.
+
+
+A list of all the test modules?
+-------------------------------
+
+With the sheer size and variety in this directory (~100 files of tests,
+covering literally every line of Hypothesis and then some), a general list
+wouldn't be that helpful!  Tips:
+
+- Check for filenames that are relevant to your contribution
+- Use ``git grep`` to search for keywords, eg the name of a strategy you've changed
+- Deliberately break something related to your code, and see which tests fail
+- Ask a maintainer!
```

Comment:
It might be worth noting here that some of the organisation is super arbitrary and that if it doesn't make sense then it might just not make sense rather than there being some missing organisational principle...